src/web/profiles.rs

```diff
@@ -0,0 +1,104 @@
+use std::{collections::HashMap, fs};
+
+use anyhow::Context as _;
+use toml_edit::DocumentMut;
+
+use crate::external_cli::{arg_builder::ArgBuilder, cargo::metadata::Metadata};
+
+/// Create `--config` args to configure the default profiles to use when compiling for the web.
+pub(crate) fn configure_default_web_profiles(metadata: &Metadata) -> anyhow::Result<ArgBuilder> {
+    let manifest = fs::read_to_string(metadata.workspace_root.join("Cargo.toml"))
+        .context("failed to read workspace manifest")?
+        .parse::<DocumentMut>()
+        .context("failed to parse workspace manifest")?;
+
+    let mut args = ArgBuilder::new();
+
+    if !is_profile_defined_in_manifest(&manifest, "web") {
+        args = args.append(configure_web_profile());
+    }
+
+    if !is_profile_defined_in_manifest(&manifest, "web-release") {
+        args = args.append(configure_web_release_profile());
+    }
+
+    Ok(args)
+}
+
+fn is_profile_defined_in_manifest(manifest: &DocumentMut, profile: &str) -> bool {
+    manifest
+        .get("profile")
+        .is_some_and(|profiles| profiles.get(profile).is_some())
+}
+
+/// Configure the default profile for web debug builds.
+///
+/// It is optimized for fast iteration speeds.
+fn configure_web_profile() -> ArgBuilder {
+    configure_profile("web", "dev", HashMap::new())
+}
+
+/// Configure the default profile for web release builds.
+///
+/// It is optimized both for run time performance and loading times.
+fn configure_web_release_profile() -> ArgBuilder {
+    let config = HashMap::from_iter([
+        // Optimize for size, greatly reducing loading times
+        ("opt-level", "s"),
+        // Remove debug information, reducing file size further
+        ("strip", "debuginfo"),
+    ]);
+    configure_profile("web-release", "release", config)
+}
```

Comment:
Instead of configuring profiles through code, would it be possible to specify the defaults in `assets/web/profiles.toml` and load those instead?

> Cargo also accepts arbitrary configuration overrides through the --config command-line option. The argument should be in TOML syntax of KEY=VALUE or **provided as a path to an extra configuration file**:
>
> ```sh
> # With a path to a configuration file
> cargo --config ./path/to/my/extra-config.toml fetch
> ```
>
> [Command-line overrides, Cargo Configuration](https://doc.rust-lang.org/cargo/reference/config.html#command-line-overrides)

Reply:
To add on to the configuration themselves, [min-sized-rust](https://github.com/johnthagen/min-sized-rust) is a great resource. I'm happy with the current config for now, but it may be worth looking into:

- `lto = "thin"` (I've had good experience with thin LTO, I feel it's the right balance between optimization and compilation speed)
- `codegen-units = 1` (Setting this to 1 _significantly_ hurts compile times, so I'm not sure we should do it by default)
- `panic = "abort"` (I'm not sure if this breaks anything within Bevy)

Reply:
> Instead of configuring profiles through code, would it be possible to specify the defaults in `assets/web/profiles.toml` and load those instead?
> 
> > Cargo also accepts arbitrary configuration overrides through the --config command-line option. The argument should be in TOML syntax of KEY=VALUE or **provided as a path to an extra configuration file**:
> > ```shell
> > # With a path to a configuration file
> > cargo --config ./path/to/my/extra-config.toml fetch
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > [Command-line overrides, Cargo Configuration](https://doc.rust-lang.org/cargo/reference/config.html#command-line-overrides)

I don't think this would be possible (without additional overhead).
Because when using `--config PATH`, the path will be relative to the current working directory.
But which path would we point to? In reality, the `assets/web/profiles.toml` file will not exist after `bevy_cli` has been installed, so we can't point to it.

I think the only way to keep the `.toml` file would be to read it to a string at compile time (like we do for `index.html`), but then we have to parse the TOML and convert it back into the key value pairs to pass to `--config`.

So at this point, I'm not sure if it's worth it.
But if you have other ideas on how to solve it let me know, storing the defaults in a `.toml` file would definitely be cool!

Reply:
No, you're right. That would be too much extra work, so I'm happy keeping it as-is!